tests/integration-tests/conftest.py

```diff
@@ -1081,11 +1086,64 @@ def vpc_stacks(cfn_stacks_factory, request):
             ],
         )
         template = NetworkTemplateBuilder(
-            vpc_configuration=vpc_config, default_availability_zone=availability_zones[0]
+            vpc_configuration=vpc_config,
+            default_availability_zone=availability_zones[0],
+            create_bastion_instance=True,
+            bastion_key_name=key_name,
+            region=region,
         ).build()
-        vpc_stacks[region] = _create_vpc_stack(request, template, region, cfn_stacks_factory)
+        vpc_stacks_dict[region] = _create_vpc_stack(request, template, region, cfn_stacks_factory)
+
+    return vpc_stacks_dict
+
+
+@pytest.fixture(scope="class")
+def vpc_stack_with_endpoints(region, cfn_stacks_factory, request, key_name):
+    """
+    Create a VPC stack with VPC endpoints.
+    Since VPC endpoints modify DNS at VPC level, all the subnets in that VPC will be affected.
+    :return: a VPC stack
+    """
+
+    credential = request.config.getoption("credential")
+    availability_zones = get_availability_zones(region, credential)  # FIXME
+
+    bastion_subnet = SubnetConfig(
+        name="Bastion",
+        cidr="192.168.32.0/20",
+        map_public_ip_on_launch=True,
+        has_nat_gateway=True,
+        availability_zone=availability_zones[0],
+        default_gateway=Gateways.INTERNET_GATEWAY,
+    )
+
+    no_internet_subnet = SubnetConfig(
+        name="NoInternet",
+        cidr="192.168.16.0/20",  # 4096 IPs
+        map_public_ip_on_launch=False,
+        has_nat_gateway=False,
+        availability_zone=availability_zones[0],
+        default_gateway=Gateways.NONE,
+    )
+
+    vpc_config = VPCConfig(
+        cidr="192.168.0.0/17",
+        additional_cidr_blocks=["192.168.128.0/17"],
+        subnets=[
+            bastion_subnet,
+            no_internet_subnet,
+        ],
+    )
+
```

Comment:
Here we are missing the subnets we added for MultiAZ: 

https://github.com/aws/aws-parallelcluster/blob/develop/tests/integration-tests/conftest.py#L1037-L1068

Reply:
the test using this stack isn't multi-az. Making the test multi-az is out of scope